src/miral/application_selector.cpp

```diff
@@ -21,243 +21,439 @@
 
 using namespace miral;
 
-ApplicationSelector::ApplicationSelector(miral::WindowManagerTools const& in_tools) :
-    tools{in_tools}
-{}
+namespace miral
+{
+
+template <typename T>
+class WindowCyclingBehavior
```

Comment:
I don't particularly like this name, if you can think of something better

Reply:
The class handles  ordering a stack of `T` items such that they are always in the correct Alt + Tab order that we want

Reply:
`TabOrderSelector`?